cli/api.py

```diff
@@ -53,8 +63,53 @@ def check_auth(id_token: str) -> Optional[str]:
     if response.status_code != 200:
         raise ApiError(_error_message(response))
 
-    # No errors, so the token is valid
-    return None
+
+def retry_with_reauth(api_request):
+    """
+    For a function `api_request` that returns a `Response` object, if that response
+    has status code 403, prompt the user to enter a fresh ID token from the portal,
+    and retry the request.
+    """
+
+    TOKEN_URL = f'https://{"staging" if get_env() != "prod" else ""}portal.cimac-network.org/assays/cli-instructions'
+
+    @wraps(api_request)
+    def wrapped(*args, **kwargs):
+        while True:
+            res = api_request(*args, **kwargs)
+            # If the error isn't auth-related, break out of the retry loop.
+            if res.status_code != 403:
+                break
+
+            # Prompt the user for a new ID token.
+            while True:
+                click.prompt(
+                    (
+                        "\nCIDC reauthentication required. Please copy a fresh identity token from the Portal "
+                        f"to your clipboard here:\n\n\t{TOKEN_URL}\n\n"
+                        "Then, press 'enter' to paste your copied token below"
+                    ),
+                    default="enter",
+                    show_default=False,
+                )
+                id_token = _read_clipboard()
```

Comment:
exception handling here? what if there's no display/clipboard for Tk to operate on?